### PR TITLE
Set table name if there are blank named case types

### PR DESCRIPTION
**Pull request by proteusvacuum**
This probably isn't the best solution since we should likely do something smarter in the data dictionary about not showing blank case types, or warning the user or something.

https://dimagi-dev.atlassian.net/browse/HI-363

### CODE REVIEW

**Review by gbova — APPROVED** [review]

File: corehq/apps/data_dictionary/views.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def _export_data_dictionary(domain):
     )
     export_data = {}
     for case_type in queryset:
-        export_data[case_type.name] = [{
+        export_data[case_type.name or "No Name"] = [{
```
Comment (orangejenny) [inline corehq/apps/data_dictionary/views.py]: translate?

**Comment by orangejenny** [comment]
What would you think of also defaulting to "No Name" in the UI, so they can see that there's a blank case type?

File: corehq/apps/data_dictionary/views.py (hunk 1 above)
Comment (proteusvacuum) [reply corehq/apps/data_dictionary/views.py]: None of the other parts of this download are translated, so I went with this. Is it worth translating the whole thing?

Comment (orangejenny) [reply corehq/apps/data_dictionary/views.py]: How about translating this function so the download is done and not worrying about the rest of the UI right now?

Comment (orangejenny) [reply corehq/apps/data_dictionary/views.py]: I think we usually do translate feature flagged content, just not admin-only content like accounting.

**Pull request merged**